src/Generator/Generator.php

```diff
@@ -867,7 +867,38 @@ public function getSoapClient()
     public function getUrlContent($url)
     {
         if (strpos($url, '://') !== false) {
-            return Utils::getContentFromUrl($url, $this->getOptionBasicLogin(), $this->getOptionBasicPassword(), $this->getOptionProxyHost(), $this->getOptionProxyPort(), $this->getOptionProxyLogin(), $this->getOptionProxyPassword(), $this->getSoapClient()->getSoapClientStreamContextOptions());
+            $content = Utils::getContentFromUrl(
+                $url, 
+                $this->getOptionBasicLogin(), 
+                $this->getOptionBasicPassword(), 
+                $this->getOptionProxyHost(), 
+                $this->getOptionProxyPort(), 
+                $this->getOptionProxyLogin(), 
+                $this->getOptionProxyPassword(), 
+                $this->getSoapClient()->getSoapClientStreamContextOptions()
+            );
+            if($this->getOptions()->getSchemasSave() === true) {
+                $schemasFolder = $this->getOptions()->getSchemasFolder();
+                if(($schemasFolder == null) || empty($schemasFolder)) {
+                  $schemasFolder = 'wsdl';
+                }
+                $directory = rtrim($this->getOptions()->getDestination(), DIRECTORY_SEPARATOR)
+                    . DIRECTORY_SEPARATOR
+                    . rtrim($schemasFolder, DIRECTORY_SEPARATOR);
+                // @todo Cover all possible variants
+                if(
+                    (strpos(strtolower($url),'.wsdl') !== false) 
+                    || (strpos(strtolower($url),'.xsd') !== false) 
+                    || (strpos(strtolower($url),'.xml') !== false)
+                ) {
+                    $filename = basename($url);
+                } else {
+                    $filename = 'schema.wsdl';
+                }
+                Utils::createDirectory($directory);
+                file_put_contents($directory . DIRECTORY_SEPARATOR . $filename, $content);
+            }
```

Comment:
The whole section needs to be formatted, did you run `composer lint` before committing your code ?

In addition, it would be better to extract the whole section into the Util class until it is clearly handled another way otherwise it makes the method heavy